### PR TITLE
Admin sees all orders and edits status types #40

### LINKED ISSUE #40: Admin Orders
As an Admin
      When I visit the dashboard
      Then I can see a listing of all orders
      And I can see the total number of orders for each status ("Ordered", "Paid", "Cancelled", "Completed")
      And I can see a link for each individual order
      And I can filter orders to display by each status type  ("Ordered", "Paid", "Cancelled", "Completed")
      And I have links to transition the status
        - I can click on "cancel" on individual orders which are "paid" or "ordered"
        - I can click on "mark as paid" on orders that are "ordered"
        - I can click on "mark as completed" on orders that are "paid"

### PR DESCRIPTION
**Pull request by brennanholtzclaw**
As an Admin
      When I visit the dashboard
      Then I can see a listing of all orders
      And I can see the total number of orders for each status ("Ordered", "Paid", "Cancelled", "Completed")
      And I can see a link for each individual order
      And I can filter orders to display by each status type  ("Ordered", "Paid", "Cancelled", "Completed")
      And I have links to transition the status
        - I can click on "cancel" on individual orders which are "paid" or "ordered"
        - I can click on "mark as paid" on orders that are "ordered"
        - I can click on "mark as completed" on orders that are "paid"

closes #40

### CODE REVIEW

File: app/controllers/orders_controller.rb [hunk 1]
```diff
@@ -22,4 +22,12 @@ def create
   def show
     @order = Order.find(params[:id])
   end
+
+  def update
+    # binding.pry
```
Comment (ksk5280) [inline app/controllers/orders_controller.rb]: remove binding.pry comment

File: app/views/admin/dashboard/index.html.erb [hunk 2]
```diff
@@ -4,4 +4,41 @@
...
+      <h2 class="green-font mousiness" >Order Statuses</h1>
+    </div>
+  </div>
+  <div class="orders row">
+    <ul>
+      <li><%= link_to "Ordered", admin_status_path(status: "ordered")%>: <%= @orders.status_total("ordered") %></li>
+      <li><%= link_to "Cancelled", admin_status_path(status: "cancelled")%>: <%= @orders.status_total("cancelled") %></li>
+      <li><%= link_to "Paid", admin_status_path(status: "paid")%>: <%= @orders.status_total("paid") %></li>
+      <li><%= link_to "Completed", admin_status_path(status: "completed")%>: <%= @orders.status_total("completed") %></li>
+    </ul>
```
Comment (ksk5280) [inline app/views/admin/dashboard/index.html.erb]: We should use an enumerable for these links

File: app/views/admin/dashboard/index.html.erb [hunk 3]
```diff
@@ -4,4 +4,41 @@
...
+        <th>Status</th>
+        <th>Placed on</th>
+        <th>Update Order</th>
+      </tr>
+      <% @orders.each do |order| %>
+        <tr>
+          <td><%= link_to "##{order.id}", order_path(order.id) %></td>
+          <td><%= order.status %></td>
+          <td><%= order.created_at.strftime("%B %-d, %Y") %></td>
+          <% if order.status == "paid" %>
```
Comment (ksk5280) [inline app/views/admin/dashboard/index.html.erb]: This logic should be pulled out into a helper method if possible

File: spec/features/admin_views_all_orders_spec.rb [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    click_on "Completed"
+
+    within ".orders" do
+      expect(page).to_not have_content "##{@order2.id}"
+      expect(page).to_not have_content "cancelled"
+      expect(page).to_not have_content "February 13, 2016"
+    end
+  end
+
+  scenario "admin tries to visit status path that does not exist" do
```
Comment (ksk5280) [inline spec/features/admin_views_all_orders_spec.rb]: I think this should be changed to something like:
context "admin tries to visit status path that does not exist" do
   scenario "sees invalid status notification and admin dashboard" do ...

File: spec/features/admin_views_all_orders_spec.rb [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+      expect(page).to have_content "February 13, 2016"
+
+      expect(page).to have_content("Ordered: 1")
+      expect(page).to have_content("Paid: 0")
+      expect(page).to have_content("Cancelled: 1")
+      expect(page).to have_content("Completed: 0")
+    end
+  end
+
+  context "admin tries to change status of a single order" do
```
Comment (ksk5280) [inline spec/features/admin_views_all_orders_spec.rb]: To be more clear, this should be something like:
  scenario "sees notification of status update" do

File: spec/features/admin_views_all_orders_spec.rb (hunk 4 above)
Comment (erinnachen) [inline spec/features/admin_views_all_orders_spec.rb]: Cool I'll edit this

File: app/views/admin/dashboard/index.html.erb (hunk 3 above)
Comment (erinnachen) [inline app/views/admin/dashboard/index.html.erb]: I'm going to leave this in for right now, it's something that will clearly get taken care of when we refactor.

Comment (ksk5280) [inline app/views/admin/dashboard/index.html.erb]: Sounds good, I created a new issue so that we'll remember to get back to it

**Pull request merged**